### ISSUE TITLE
can the network be trained?

### ISSUE THREAD

**Issue by KungFuPandaPro** [qcf-568/DocTamper#4]
I use your network from class seg_dtd to train my dataset,but it is abnormal,it can not convergence.
![企业微信截图_16950020521493](https://github.com/qcf-568/DocTamper/assets/67446107/fb8aa766-895f-400d-a697-78ef16724c04)

**Comment by qcf-568** [comment]
The model can convergence if you train it in a correct manner.
Could you please provide more details about your dataset and data pipeline, so maybe I can help you identify the problem?

**Comment by KungFuPandaPro** [comment]
> The model can convergence if you train it in a correct manner. Could you please provide more details about your dataset and data pipeline, so maybe I can help you identify the problem?
![企业微信截图_16950160601720](https://github.com/qcf-568/DocTamper/assets/67446107/4693bd4f-f934-441f-a8ec-0be6e1fa6f7c)
![企业微信截图_16950160765131](https://github.com/qcf-568/DocTamper/assets/67446107/ce780b01-a5a0-4da7-924b-c630ef7cc81a)
thank you for your repley,this is my dataloar.And my datasets are some id cards.when zero in label is background,one in label is positive.

**Comment by qcf-568** [comment]
I think the function "get_specilized_shape_img" is the reason why the training process doesn't convergence.
I suggest not to resize the img during training, just cutting them to random patches of fixed shape (or padding on the right and bottom if necessary), and make sure that the starting point coordinates of the patches are integer multiples of eight.
(e.g. ori_img of shape (h=1111,w=2222), do not resize it, pad it 1 pixel on the right and 2 pixel on the bottom to (h=1112,w=2224) so they can be divided by 8, cut random patches img[sx:sx+512, sy:sy+512], where sx = random.randint(0, (h-512)//8)*8, sy=random.randint(0,(w-512)//8)*8) )

**Comment by KungFuPandaPro** [comment]
> I think the function "get_specilized_shape_img" is the reason why the training process doesn't convergence. I suggest not to resize the img during training, just cutting them to random patches of fixed shape (or padding on the right and bottom if necessary), and make sure that the starting point coordinates of the patches are integer multiples of eight. (e.g. ori_img of shape (h=1111,w=2222), do not resize it, pad it 1 pixel on the right and 2 pixel on the bottom to (h=1112,w=2224) so they can be divided by 8, cut random patches img[sx:sx+512, sy:sy+512], where sx = random.randint(0, (h-512)//8)*8, sy=random.randint(0,(w-512)//8)*8) )

I had tried,but did not work,I had not load the pretrain model you provide,is it the reason of that?

**Comment by qcf-568** [comment]
Imagenet pretrained weights are necessary.

**Comment by KungFuPandaPro** [comment]
> Imagenet pretrained weights are necessary.

But your network in cuda inference result type is torch.float16,I need to transform the result to torch.float32 before compute loss.I supppose that is the reason, how can I that the output type in cuda  become torch.float32?

**Comment by qcf-568** [comment]
The training framework in the previous project of the area used float16, I just followed it directly in my work. It works well and I don't think that is the reason.

**Comment by KungFuPandaPro** [comment]
The network is trained first epoch the loss is  

> The training framework in the previous project of the area used float16, I just followed it directly in my work. It works well and I don't think that is the reason.

I set the lr to be lower and it works a little ,but still hard to train that after 3 epoch it crush,loss become larger.I want  to test the trained model now,how to process img when validate? pad the image with neareast shape,such as [256,256],[512.512],[1024,1024] or crop the img to many block to pred ?

**Comment by qcf-568** [comment]
If the images before tampering are photographed images, and the model follows a correct training pipeline, I have never encountered a case that model does not converge.
